### PR TITLE
chore(qe): update mobc to include importcjj/mobc#86

### PR DESCRIPTION
**Pull request by miguelff**
Fixes https://github.com/prisma/prisma/issues/21221

### CODE REVIEW

**Comment by miguelff** [comment]
Created as an integration branch, in case some of the tests in prisma/prisma depended on the previous (wrong) behavior described in https://github.com/importcjj/mobc/issues/82

**Comment by miguelff** [comment]
According to https://github.com/prisma/prisma/pull/21591 there aren't failing tests after upgrading mobc.

**Comment by Jolg42** [comment]
@miguelff it would be interesting to review tests, we should have had one catching this change ideally, right?

**Comment by miguelff** [comment]
I don't want to encode in the tests  specific values for queries, because depending on pool sizes, those will behave differently. What I observed is that  before (main) reported idle = pool size, while now, it reports 1. 

This can sometimes be reported as 1 or 0, and lead to flakey tests, tho' as metrics are reported asynchronously.

[main.txt](https://github.com/prisma/prisma-engines/files/13336783/main.txt)
[branch.txt](https://github.com/prisma/prisma-engines/files/13336785/branch.txt)

Screenshot 2023-11-13 at 17 09 24

**Comment by Jolg42** [comment]
Right!

My thinking is that we could expect 0 or 1, I think that's better than nothing. Though not ideal.

File: query-engine/black-box-tests/tests/metrics/smoke_tests.rs [hunk 1]
```diff
@@ -14,6 +15,24 @@ mod smoke_tests {
         schema.to_owned()
     }
 
+    fn assert_value_in_range(metrics: &str, metric: &str, low: f64, high: f64) {
+        let regex = Regex::new(format!(r"{metric}\s+([+-]?\d+(\.\d+)?)").as_str()).unwrap();
```
Comment (janpio) [inline query-engine/black-box-tests/tests/metrics/smoke_tests.rs:19]: We can also return metrics as Json, that would probably simplify the whole test here :D 
(I know, too late - you already wrote the code)

**Review by janpio — APPROVED** [review]

**Pull request merged**